Insight_Project_Framework/Classifier.py

```diff
@@ -0,0 +1,123 @@
+        page. num_pages limit number of pages to read in.
+    """
+
+    def __init__(self, text_folder_path, num_pages, language, keys, county_names, num_features, clf, test_ratio):
+        """Initialize variables for Classifier class."""
+        self.text_folder_path = text_folder_path
+        self.num_pages = num_pages # number of pages to process
+        self.language = language
+        self.keys = keys # county keywords
+        self.county_names = county_names
+        self.word_processor = Word_processor(language, num_pages, county_names, text_folder_path)
+        self.num_features = num_features # dimension of features
+        self.feature_vectorizer = TfidfVectorizer(max_features=self.num_features)
+        self.le = preprocessing.LabelEncoder()
+        self.clf = clf
+        self.test_ratio = test_ratio
+
+    def preprocess_fun(self):
+        """
+        Process and prepare data for training.
+        
+        Loads county names, and add them into the lexicon.
+        Load and clean the training data.
+        Convert texts into TF-IDF vectors.
+        Train-test split the data.
+        """
+        self.word_processor.lexicon_add_words(self.county_names)
+        # load data and clean
+        self.word_processor.load_training_data_and_clean()
+        # convert string labels to numerical
+        self.encoded_labels = self.le.fit_transform(self.word_processor.labels)
+        self.tf_idf_vector = self.feature_vectorizer.fit_transform(self.word_processor.corpus)
+        # train test split
+        self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(self.tf_idf_vector.toarray(), self.encoded_labels, test_size=self.test_ratio, stratify=self.encoded_labels)
+        print("Training data processed.")
+
+    def simple_inference(self):
+        """
+        Make predictions using simple rules.
+
+        If one and only one county name found in processed text, return the name as 
+        prediction. If no or more than 1 county name found from text, consider the system unable
+        to make a prediction. This approach is used as a baseline.
+        """
+        correct_preds = 0
+        result_DF = pd.DataFrame(np.zeros((len(self.word_processor.corpus), len(self.keys))), columns = self.county_names)
+        for i in range(len(self.word_processor.corpus)):
+            doc = self.word_processor.corpus[i]
+            for j in range(len(self.county_names)):
+                county_name = self.county_names[j]
+                if county_name in doc:
+                    result_DF.loc[i, county_name] = 1
+
+        for i in range(len(result_DF)):
+            if np.sum(result_DF.loc[i]) == 1 and np.where(result_DF.loc[i] == 1)[0][0] == self.encoded_labels[i]:
+                correct_preds += 1
+        print("baseline accuracy: " + str(correct_preds/len(self.encoded_labels)))
+
+    def train(self):
+        self.clf.fit(self.X_train, self.y_train)
```

Comment:
You may want to add logic to handle errors and exceptions or to add functionality to log outputs and errors instead of relying on print. For this project, it is fine to leave it as is.